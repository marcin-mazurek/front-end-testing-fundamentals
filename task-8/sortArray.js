export default function sortArray(array) {
  if (!array || !array instanceof Array) {
    throw new TypeError('Invalid argument - expected a function');
  }

  var minIdx, temp, len = array.length;

  for (var i = 0; i < len; i++) {
    minIdx = i;

    for (var j = i+1; j < len; j++) {
      if (array[j] < array[minIdx]) {
        minIdx = j;
      }
    }

    temp = array[i];
    array[i] = array[minIdx];
    array[minIdx] = temp;
  }
  
  return array;
}