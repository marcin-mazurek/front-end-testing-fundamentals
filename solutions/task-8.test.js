import sortArray from '../task-8/sortArray';

describe('sortArray', () => {
  it('sorts an array correctly', () => {
    expect(sortArray([0, 808, 6, 3, 55, 102, 2, 34, 1010, 3, 42, 14, 61])).toMatchSnapshot();
    expect(sortArray(['lorem', 'ipsum', 'dolor', 'sit', 'amet'])).toMatchSnapshot();
  });

  it('throws when incorrect data type is passed as argument', () => {
    expect(() => sortArray(null)).toThrowErrorMatchingSnapshot();
  });
});