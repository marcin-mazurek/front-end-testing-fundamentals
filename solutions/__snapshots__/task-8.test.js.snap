// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortArray sorts an array correctly 1`] = `
Array [
  0,
  2,
  3,
  3,
  6,
  14,
  34,
  42,
  55,
  61,
  102,
  808,
  1010,
]
`;

exports[`sortArray sorts an array correctly 2`] = `
Array [
  "amet",
  "dolor",
  "ipsum",
  "lorem",
  "sit",
]
`;

exports[`sortArray throws when incorrect data type is passed as argument 1`] = `"Invalid argument - expected a function"`;
